// Copyright (C) 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package icing.lib;

import "icing/proto/status.proto";

option java_package = "com.google.android.icing.proto";
option java_multiple_files = true;
option objc_class_prefix = "ICNG";

// Defines the blob operation result proto that user try to read/write/commit a
// blob to Icing.
//
// Next tag: 3
message BlobProto {
  // Status code can be one of:
  //   OK
  //   FAILED_PRECONDITION
  //
  // See status.proto for more details.
  //
  optional StatusProto status = 1;

  // The file decriptor of the blob file from Icing.
  optional int32 file_descriptor = 2;
}
